auto-increment.md

```diff
@@ -101,53 +101,137 @@ In the example above, perform the following operations in order:
+| a       | b                   |
++---------+---------------------+
+|       1 | 2020-09-09 20:38:22 |
+|       2 | 2020-09-09 20:38:22 |
+|       3 | 2020-09-09 20:38:22 |
+| 2000001 | 2020-09-09 20:43:43 |
+|       4 | 2020-09-09 20:44:43 |
+| 2030001 | 2020-09-09 20:54:11 |
++---------+---------------------+
+6 rows in set (0.00 sec)
+```
 
-### Relation with `_tidb_rowid`
+A high rate of TiDB server restarts may contribute to `AUTO_INCREMENT` values being exhausted. In the above example, the TiDB server still had values `[5-30000]` free in its cache. These values are lost, and will not be reallocated.
 
-If the primary key of integer type does not exist, TiDB uses `_tidb_rowid` to identify rows. `_tidb_rowid` and the auto-increment column (if any) share an allocator. In such cases, the cache size might be consumed by both `_tidb_rowid` and the auto-increment column. Therefore, you might encounter the following situation:
+It is not recommended to rely on`AUTO_INCREMENT` values being continuous. Consider the following example, where a TiDB server has a cache of values `[2000001-2030000]`. By manually inserting the value `2029998` we can see the behavior as a new cache range is retrieved:
 
 ```sql
-mysql> create table t(id int unique key AUTO_INCREMENT);
-Query OK, 0 rows affected (0.05 sec)
+mysql> INSERT INTO t (a) VALUES (2029998);
+Query OK, 1 row affected (0.01 sec)
 
-mysql> insert into t values (),(),();
-Query OK, 3 rows affected (0.00 sec)
-Records: 3  Duplicates: 0  Warnings: 0
+mysql> INSERT INTO t (a) VALUES (NULL);
+Query OK, 1 row affected (0.01 sec)
+
+mysql> INSERT INTO t (a) VALUES (NULL);
+Query OK, 1 row affected (0.00 sec)
+
+mysql> INSERT INTO t (a) VALUES (NULL);
+Query OK, 1 row affected (0.02 sec)
 
-mysql> select _tidb_rowid, id from t;
-+-------------+------+
-| _tidb_rowid | id   |
-+-------------+------+
-|           4 |    1 |
-|           5 |    2 |
-|           6 |    3 |
-+-------------+------+
-3 rows in set (0.01 sec)
+mysql> INSERT INTO t (a) VALUES (NULL);
+Query OK, 1 row affected (0.01 sec)
+
+mysql> SELECT * FROM t ORDER BY b;
++---------+---------------------+
+| a       | b                   |
++---------+---------------------+
+|       1 | 2020-09-09 20:38:22 |
+|       2 | 2020-09-09 20:38:22 |
+|       3 | 2020-09-09 20:38:22 |
+| 2000001 | 2020-09-09 20:43:43 |
+|       4 | 2020-09-09 20:44:43 |
+| 2030001 | 2020-09-09 20:54:11 |
+| 2029998 | 2020-09-09 21:08:11 |
+| 2029999 | 2020-09-09 21:08:11 |
+| 2030000 | 2020-09-09 21:08:11 |
+| 2060001 | 2020-09-09 21:08:11 |
+| 2060002 | 2020-09-09 21:08:11 |
++---------+---------------------+
+11 rows in set (0.00 sec)
 ```
 
+After the value `2030000` is inserted, the next value is `2060001`. This jump in sequence is due to another TiDB server obtaining the intermediate cache range of `[2030001-2060000]`. When multiple TiDB servers are deployed, there will be gaps in the `AUTO_INCREMENT` sequence due to cache requests being interleaved.
```

Comment:
```suggestion
After the value `2030000` is inserted, the next value is `2060001`. This jump in sequence is due to another TiDB server obtaining the intermediate cache range of `[2030001-2060000]`. When multiple TiDB servers are deployed, there will be gaps in the `AUTO_INCREMENT` sequence because cache requests are interleaved.
```